cmake/sdp.cmake

```diff
@@ -4,6 +4,24 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+function(sdp_assembly hrt_srcs)
+  sdp_assembly_generate("${CMAKE_SOURCE_DIR}/${hrt_srcs}")
+  sdp_assembly_check("${CMAKE_SOURCE_DIR}/${hrt_srcs}")
+  sdp_assembly_prepare_install("${CMAKE_SOURCE_DIR}/${hrt_srcs}")
+  sdp_assembly_target_sources("${CMAKE_SOURCE_DIR}/${hrt_srcs}")
+
+  add_dependencies(app asm_check)
```

Comment:
Functions should not be hardcoded to a given target / library but take this as a function argument.